text/0000-full-wallet-lifecycle.md

```diff
@@ -0,0 +1,165 @@
+to initialize the wallet manually before the APIs can be used, or provide custom management for wallet lifecycle functions.
+
+The Wallet APIs are intended to be the foundation upon which community-created wallets should be built, and the job of a wallet creator is made far
+more difficult by the absence of wallet creation and seed management functions within the API. Ideally, it should be the case that a wallet can
+be instantiated and managed solely via the Owner API.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+From an end-user perspective, (i.e. end-users of community wallets that use the wallet API,) this change should be transparent.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Wallet Initialization
+
+Currently, wallet data does not exist until the user runs `grin-wallet init`. The `init` command creates `grin-wallet.toml`,
+in the `~/.grin/main` directory (or `~/.grin/floonet`, or the current directory via the `-h` flag), prompts the user for a password,
+creates a seed file, stores the resulting data files in the directory specified in `grin-wallet.toml` (`~/.grin/main/wallet_data` by default)
+and initialises the lmdb database.
+
+It should be possible to run `grin-wallet owner_api` or invoke the API directly from a linked binary without having instantiated a wallet.
+
+## Security Model
+
+Given that the Wallet's Owner API needs to deal with sensitive data such as passwords and seed phrases, the API will be enhanced with a new ECDH and Token-based security model, the primary goals of which are to:
+
+* Ensure sensitive data such as passwords or seed phrases are always end-to-end encrypted between the client and Owner API server, regardless of what higher-level protocols are used during the exchange.
+* Minimize the potential for damage that can be done by a third party listening on the exchange between a wallet client and its corresponding server.
+* Ensure that sensitive data such as passwords or seed phrases are not resident in server-side memory any longer than they absolutely need to be.
+
+Note that this mode of operation is primarly intended for use over the JSON-RPC API, which supports many different architectural possiblities. Clients that link libraries directly and keep all sensitive data in the same process would see less benefit from this scheme, and an alternative model which doesn't encrypt any sensitive data is provided. Further, authors of existing wallets will need time to consider and/or implement the added complexity needed on the client-side to support ECDH and encryption. It's therefore proposed that the Owner API initially provide the new "SecureAPI" mode as an optional feature, with wallet authors strongly encouraged to make use of it. Support for the "InsecureAPI" model can be maintained indefinitely for directly-linked wallets, and for the JSON-RPC API until a cut-off release at some point in the future.
```

Comment:
> Note that this mode of operation is primarly intended

Not English native but isn't it primarily?

Also

> architectural possiblities

architectural possibilities.






Missing "s" at:
> It's therefore proposed that the Owner API initially provide